torchbenchmark/models/dcgan/__init__.py

```diff
@@ -0,0 +1,380 @@
+            nn.BatchNorm2d(dcgan.ngf),
+            nn.ReLU(True),
+            # state size. (dcgan.ngf) x 32 x 32
+            nn.ConvTranspose2d( dcgan.ngf, dcgan.nc, 4, 2, 1, bias=False),
+            nn.Tanh()
+            # state size. (dcgan.nc) x 64 x 64
+        )
+
+        self.jt = None
+        self.jitshape = None
+        self.debug_print = False
+
+    def forward(self, input):
+
+      if self.debug_print:
+        print(input.shape)
+
+      # Not enabled
+      if False:
+        if self.jitshape != input.shape:
+            self.jitshape = input.shape
+
+            print(input.shape)
+            self.jt = torch.jit.trace(self.main, input)
+
+            print("GEN BEFORE")
+            print(self.jt.graph)
+            
+            self.jt = torch.quantization.fuse_modules(self.jt, [['bn', 'relu']], inplace=False)
+
+
+            print("GEN AFTER")
+            print(self.jt.graph)
+
+        return self.jt(input)
+
+      return self.main(input)
+
+class Discriminator(nn.Module):
+    def __init__(self, ncgan):
+        ngpu = ncgan.ngpu
+        nc = ncgan.nc
+        ndf = ncgan.ndf
+
+        super(Discriminator, self).__init__()
+        self.ngpu = ngpu
+        self.main = nn.Sequential(
+            # input is (nc) x 64 x 64
+            nn.Conv2d(nc, ndf, 4, 2, 1, bias=False),
+            nn.LeakyReLU(0.2, inplace=True),
+            # state size. (ndf) x 32 x 32
+            nn.Conv2d(ndf, ndf * 2, 4, 2, 1, bias=False),
+            nn.BatchNorm2d(ndf * 2),
+            nn.LeakyReLU(0.2, inplace=True),
+            # state size. (ndf*2) x 16 x 16
+            nn.Conv2d(ndf * 2, ndf * 4, 4, 2, 1, bias=False),
+            nn.BatchNorm2d(ndf * 4),
+            nn.LeakyReLU(0.2, inplace=True),
+            # state size. (ndf*4) x 8 x 8
+            nn.Conv2d(ndf * 4, ndf * 8, 4, 2, 1, bias=False),
+            nn.BatchNorm2d(ndf * 8),
+            nn.LeakyReLU(0.2, inplace=True),
+            # state size. (ndf*8) x 4 x 4
+            nn.Conv2d(ndf * 8, 1, 4, 1, 0, bias=False),
+            nn.Sigmoid()
+        )
+        self.jt = None
+        self.jitshape = None
+
+    def forward(self, input):
+      if False:
+        if self.jitshape != input.shape:
+            self.jitshape = input.shape
+            print(input.shape)
+
+            self.jt = torch.jit.trace(self.main, input)
+
+            #self.jt = torch.quantization.fuse_modules(self.jt, [['bn', 'relu']], inplace=False)
+        
+            print(self.jt)
+
+        return self.jt(input)
+
+        if self.jt != None:
+            return self.jt(input)
+
+      return self.main(input)
+
+class Model(BenchmarkModel):
+    task = COMPUTER_VISION.GENERATION
+
+    def __init__(self, device=None, jit=False):
+        super().__init__()
+
+        self.debug_print = False
+
+        self.jit = jit
+        self.device = device
+        self.root = str(Path(__file__).parent)
+        self.jit = jit
```

Comment:
Why are we assigning `self.jit = jit` twice here? Also, if this model is not supporting JIT, please return `NotImplementedError` for JIT test.

Reply:
Jit works now.